compiler/cgen.nim

```diff
@@ -480,6 +480,10 @@ proc resetLoc(p: BProc, loc: var TLoc) =
       # on the bytes following the m_type field?
       genObjectInit(p, cpsStmts, loc.t, loc, constructObj)
 
+proc containsImportedCppType(typ: PType): bool =
+  let actualType = typ.skipTypes({tyRef, tyPtr})
```

Comment:
Why? a ptr pointing to an imported C++ type can still be set to nil safely and should be.

Reply:
Agreed, apparently, there is no need for skipping tyPtr. This was not about a pointer's operations, but to make sure that imported C++ types are actually correctly dug up by this method (i. e. if I were to remove the skipping of tyRef, ARC/ORC managed structures containing C++ classes would fall through the search pattern).